tests/Tests.FeatureManagement/FeatureManagement.cs

```diff
@@ -961,6 +963,49 @@ public async Task BindsFeatureFlagSettings()
             Assert.True(called);
         }
 
+        [Fact]
+        public async Task TelemetryPublishing()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            TestTelemetryPublisher testPublisher = new TestTelemetryPublisher();
+
+            services
+                .AddSingleton(config)
+                .AddSingleton<ITelemetryPublisher>(testPublisher)
+                .AddFeatureManagement()
+                .AddFeatureFilter<TimeWindowFilter>();
+
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IFeatureManager featureManager = serviceProvider.GetRequiredService<IFeatureManager>();
+
+            // Test a feature with telemetry disabled
+            bool result = featureManager.IsEnabledAsync(OnFeature).Result;
+
+            Assert.True(result);
```

Comment:
Would it be better to use Assert.True(await featureManager.IsEnabledAsync(OnFeature))